client/allocrunner/taskrunner/task_runner_test.go

```diff
@@ -2021,3 +2021,34 @@ func testWaitForTaskToStart(t *testing.T, tr *TaskRunner) {
 		require.NoError(t, err)
 	})
 }
+
+// TestTaskRunner_BaseLabels tests that the base labels for the task metrics
+// are set appropriately.
+func TestTaskRunner_BaseLabels(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+
+	alloc := mock.BatchAlloc()
+	alloc.Namespace = "not-default"
+	task := alloc.Job.TaskGroups[0].Tasks[0]
+	task.Driver = "raw_exec"
+	task.Config = map[string]interface{}{
+		"command": "whoami",
+	}
+
+	config, cleanup := testTaskRunnerConfig(t, alloc, task.Name)
+
+	tr, err := NewTaskRunner(config)
+	require.NoError(err)
+	defer cleanup()
```

Comment:
Not such a big deal here, but generally, cleanup invoked in defer as soon as function is called to ensure it's called.  Here if `NewTaskRunner` returned an error for example, `require.NoError(..)` will terminate test without `cleanup` being called.

Reply:
good point. there are a number of other incidents of this faux pas in the test, i will clean them up as well.